js/base/Exchange.js

```diff
@@ -1677,6 +1668,182 @@ module.exports = class Exchange {
+                'CRO': { 'CRC20': 'CRONOS' },
+            },
+            // this list is for common reserved CCXT unified network codes (some of them might have multiple aliases)
+            'unifiedNetworkCodesAndAliases': {
+                'BTC': [ 'BITCOIN' ],
+                'BEP20': [ 'BSC' ],
+                'ERC20': [ 'ETH' ],
+                'TRC20': [ 'TRX', 'TRON' ],
+                'APTOS': [],
+                'ALGORAND': [],
+                'CHILIZ': [],
+                'AVALANCHEC': [ 'AVAXC' ],
+                'AVALANCHEX': [ 'AVAXX' ],
+                'CARDANO': [ 'ADA' ],
+                'LTC': [ 'LITECOIN' ],
+                'WAVES': [],
+                'ARBITRUM': [],
+                'OPTIMISM': [],
+                'CRC20': [ 'CRONOS' ],
+                'HRC20': [ 'HECO' ],
+                // 'BNB': 'BEP2', // BNB is risky, as some exchanges are undeliberately calling 'BNB' network for BINANCE smart chain
+                'DOGECOIN': [ 'DOGE' ],
+                'SOLANA': [ 'SOL' ],
+                'POLYGON': [ 'MATIC' ],  // tbd PRC20
+                'OKC': [ 'OKX' ],
+                'COSMOS': [ 'ATOM' ],
+                'POLKADOT': [ 'DOT' ],
+                'TERRA': [ 'TERRA2' ],
+                'TERRACLASSIC': [],
+                'THEOPENNETWORK': [ 'TON' ],
+                'THORCHAIN': [ 'RUNE' ],
+                'BITCOINCASH': [ 'BCH' ],
+                'BITCOINCASHSV': [ 'BSV' ],
+                'ECASH': [ 'XEC' ],
+                'ZCASH': [ 'ZEC' ],
+            },
+            // below field needs to be s set to `true` for some exceptional exchanges. Setting it to `true` means that network ID-to-CODE relation defined in `networks|netwroksById` was done by common exchange-specific network-name (i.e. Erc-20) instead of the actual network-id (i.e. usdterc20), becuase in such case each currency has unique exchange-specific network-id (which is impossible to be pre-defined in `options`) and within fetchCurrencies() we set them automatically through '_chainIdsByNames' && '_namesByChainIds'. To see examples, check OKX/HUOBI implementations
+            'hasUniqueNetworkIds': false,
+            // below field will contain automatically generated network id/code/name mappings (users are not meant to interact with it directly, and it will be moved into a class-wide property later)
+            '_networkData': {
+                // so, in case of unique networks ids per currency, we will have three different entities: 1) unified code i.e. 'ERC20' 2) exchange specific network name i.e. 'Erc-20'  3) currency specific network-id i.e. 'usdterc20'
+                '_chainIdsByNames': {},
+                '_namesByChainIds': {},
+                '_namesByCodes': {},
+                // other data-containers
+                '_codesByIds': {},
+                '_networkCodesConflicts': {},
+            },
+            'networkCodesConflictsApproved': {}, // this can be set be user
```

Comment:
do we need this? seems to be adding complexity and I'm not sure if users will benefit much from it

Reply:
> do we need this? seems to be adding complexity and I'm not sure if users will benefit much from it

I think we need, Igor himself suggested https://discord.com/channels/690203284119617602/921046068555313202/1054815746724745266 and seems a correct way to do things with conflicting situations.

Reply:
also, it doesn't add complexity to any occasions, unless there is a conflict (which yeah, needs _that complexity_ ).

Reply:
after revision, i think this might be better to have than not have, because user submitting i.e. `CRC20` as networkCode, should see this message, because it conflicts with network-id and it should need explicit approval from user to avoid misunderstaning.
on the other side, I understand that user submitting networkCode, should also be aware that he is using unified networkcode, so CRC20 would refer to Cronos. 

but anyway, not clear situation and needs decision.

@kroitor this also needs your attention, just mention your thoughts.

Reply:
I would suggest that we do not add this, cause it seems redundant.